source/common/http/http1/legacy_parser_impl.cc

```diff
@@ -0,0 +1,209 @@
+    parser_.allow_chunked_length = 1;
+  }
+
+  Impl(http_parser_type type, void* data) : Impl(type) {
+    parser_.data = data;
+    settings_ = {
+        [](http_parser* parser) -> int {
+          auto* conn_impl = static_cast<ParserCallbacks*>(parser->data);
+          auto status = conn_impl->onMessageBegin();
+          return conn_impl->setAndCheckCallbackStatus(std::move(status));
+        },
+        [](http_parser* parser, const char* at, size_t length) -> int {
+          auto* conn_impl = static_cast<ParserCallbacks*>(parser->data);
+          auto status = conn_impl->onUrl(at, length);
+          return conn_impl->setAndCheckCallbackStatus(std::move(status));
+        },
+        nullptr, // on_status
+        [](http_parser* parser, const char* at, size_t length) -> int {
+          auto* conn_impl = static_cast<ParserCallbacks*>(parser->data);
+          auto status = conn_impl->onHeaderField(at, length);
+          return conn_impl->setAndCheckCallbackStatus(std::move(status));
+        },
+        [](http_parser* parser, const char* at, size_t length) -> int {
+          auto* conn_impl = static_cast<ParserCallbacks*>(parser->data);
+          auto status = conn_impl->onHeaderValue(at, length);
+          return conn_impl->setAndCheckCallbackStatus(std::move(status));
+        },
+        [](http_parser* parser) -> int {
+          auto* conn_impl = static_cast<ParserCallbacks*>(parser->data);
+          auto statusor = conn_impl->onHeadersComplete();
+          return conn_impl->setAndCheckCallbackStatusOr(std::move(statusor));
+        },
+        [](http_parser* parser, const char* at, size_t length) -> int {
+          static_cast<ParserCallbacks*>(parser->data)->bufferBody(at, length);
+          return 0;
+        },
+        [](http_parser* parser) -> int {
+          auto* conn_impl = static_cast<ParserCallbacks*>(parser->data);
+          auto status = conn_impl->onMessageComplete();
+          return conn_impl->setAndCheckCallbackStatusOr(std::move(status));
+        },
+        [](http_parser* parser) -> int {
+          // A 0-byte chunk header is used to signal the end of the chunked body.
+          // When this function is called, http-parser holds the size of the chunk in
+          // parser->content_length. See
+          // https://github.com/nodejs/http-parser/blob/v2.9.3/http_parser.h#L336
+          const bool is_final_chunk = (parser->content_length == 0);
+          static_cast<ParserCallbacks*>(parser->data)->onChunkHeader(is_final_chunk);
+          return 0;
+        },
+        nullptr // on_chunk_complete
+    };
+  }
+
+  RcVal execute(const char* slice, int len) {
+    return {http_parser_execute(&parser_, &settings_, slice, len), HTTP_PARSER_ERRNO(&parser_)};
+  }
+
+  void resume() { http_parser_pause(&parser_, 0); }
+
+  ParserStatus pause() {
+    http_parser_pause(&parser_, 1);
+    // http_parser pauses through http_parser_pause above.
+    return ParserStatus::Success;
```

Comment:
is this return value in preparation for the new parser?

Reply:
Yep exactly. New parser doesn't pause through an API call but through a callback rval.